src/Form/DataTransformer/ModelToIdPropertyTransformer.php

```diff
@@ -131,16 +155,16 @@ public function transform($entityOrCollection)
                 );
             }
         } else {
-            if (substr(\get_class($entityOrCollection), -1 * \strlen($this->className)) === $this->className) {
-                $collection = [$entityOrCollection];
-            } elseif ($entityOrCollection instanceof \ArrayAccess) {
+            if (substr(\get_class($value), -1 * \strlen($this->className)) === $this->className) {
+                $collection = [$value];
+            } elseif ($value instanceof \ArrayAccess) {
```

Comment:
shouldn't it be `\is_array($value) || $value instanceof \ArrayAccess` ?

So it seems like we doing the same checks in the `this->multiple` case and the not multiple one. We could definitely improve the conditions.